float64 s1
float64 s2
